public/app/features/alerting/unified/RuleViewer.tsx

```diff
@@ -183,33 +197,49 @@ export function RuleViewer({ match }: RuleViewerProps) {
           <RuleDetailsMatchingInstances rule={rule} pagination={{ itemsPerPage: DEFAULT_PER_PAGE_PAGINATION }} />
         </div>
       </RuleViewerLayoutContent>
-      {!isFederatedRule && data && Object.keys(data).length > 0 && (
-        <>
-          <div className={styles.queriesTitle}>
-            Query results <PanelChromeLoadingIndicator loading={isLoading(data)} onCancel={() => runner.cancel()} />
+      <Collapse
+        label="Query & Results"
+        isOpen={expandQuery}
+        onToggle={setExpandQuery}
+        loading={data && isLoading(data)}
+        collapsible={true}
+        className={styles.collapse}
+      >
+        {isGrafanaRulerRule(rule.rulerRule) && !isFederatedRule && (
+          <GrafanaRuleQueryViewer
+            condition={rule.rulerRule.grafana_alert.condition}
+            queries={queries}
+            evalDataByQuery={data}
+            evalTimeRanges={evaluationTimeRanges}
+            onTimeRangeChange={onQueryTimeRangeChange}
+          />
+        )}
+
+        {!isGrafanaRulerRule(rule.rulerRule) && !isFederatedRule && data && Object.keys(data).length > 0 && (
+          <div className={styles.queries}>
+            {queries.map((query) => {
```

Comment:
I know that probably having an array this is the only way, but can we have more that 1 query in here?

Reply:
No, there should be only one query for cloud rules